pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -177,14 +173,53 @@ func (w *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
 			return nil
 		}
 
-		// podOptions.namespaceSelector
+		// Backwards compatibility: podOptions.namespaceSelector
 		nsSelector, err := metav1.LabelSelectorAsSelector(w.namespaceSelector)
 		if err != nil {
 			return fmt.Errorf("failed to parse namespace selector: %w", err)
 		}
 		if !nsSelector.Matches(labels.Set(ns.GetLabels())) {
 			return nil
 		}
+
+		// Do not suspend a Pod whose owner is already managed by Kueue
+		if owner := metav1.GetControllerOf(pod.Object()); owner != nil {
+			if jobframework.IsOwnerManagedByKueue(owner) {
+				return nil
+			}
+			// Special case for Deployment integration which intentionally does not register an IsManagingObjectsOwner
+			// function so that it can use the standaloneJob path in the GenericReconciler for its Pods.
+			if owner.Kind == "ReplicaSet" && owner.APIVersion == "apps/v1" {
+				rs := &appsv1.ReplicaSet{}
+				err := w.client.Get(ctx, client.ObjectKey{Name: owner.Name, Namespace: pod.pod.GetNamespace()}, rs)
+				if err != nil {
+					return fmt.Errorf("failed to get replicaset: %w", err)
+				}
+				owner = metav1.GetControllerOf(rs)
+				if owner != nil {
+					if owner.Kind == "Deployment" && owner.APIVersion == "apps/v1" && jobframework.IsIntegrationEnabled("deployment") {
```

Comment:
Can we move this out to the same level where StatefulSet is considered, for symmetry. Then move the comment about the Special case for Deployment down just above the if. I also believe we can eliminate the special cases alltogether by the lookup as suggested in the other comment.